src/metabase/query_processor/parameters.clj

```diff
@@ -10,18 +10,116 @@
             [metabase.util :as u])
   (:import (org.joda.time DateTimeConstants DateTime)))
 
+(defn- day-range
+  [^DateTime end ^DateTime start]
+  {:end   end
+   :start start})
 
-(def ^:private ^:const relative-dates
-  #{"today"
-    "yesterday"
-    "past7days"
-    "past30days"
-    "thisweek"
-    "thismonth"
-    "thisyear"
-    "lastweek"
-    "lastmonth"
-    "lastyear"})
+(defn- week-range
+  [^DateTime end ^DateTime start]
```

Comment:
I think it makes sense to reverse the order of the fn arguments.

`range`-type functions usually have arguments like `(start, end)` as opposed to `(end, start)`, (not just in Clojure, but in other languages as well).

for example Clojure's range function is `start, end`: 

```clojure
(range 0 10) ; return numbers 0 through 10 (not inclusive)
```

Reversing the arguments here and in the other functions makes sense because that's what people will expect and we want to try to prevent people from making mistakes because they assume the function works the way other ones do. The idea here is to make it harder for people to do the wrong thing